coderd/database/migrations/000389_telemetry_heartbeats.up.sql

```diff
@@ -0,0 +1,33 @@
+CREATE TABLE telemetry_heartbeats (
+    event_type TEXT NOT NULL CONSTRAINT telemetry_heartbeat_event_type_check CHECK (event_type IN ('aibridge_interceptions_snapshot')),
+    heartbeat_timestamp TIMESTAMP WITH TIME ZONE NOT NULL,
+
+    PRIMARY KEY (event_type, heartbeat_timestamp)
+);
+
+COMMENT ON TABLE telemetry_heartbeats IS 'Telemetry heartbeat tracking table for deduplication of event types across replicas.';
+COMMENT ON COLUMN telemetry_heartbeats.event_type IS 'The type of event that was sent.';
+COMMENT ON COLUMN telemetry_heartbeats.heartbeat_timestamp IS 'The timestamp of the heartbeat event. Usually the end of the period for which the event contains data.';
```

Comment:
I think we need a better term than timestamp (or `created_at`) based on this comment.

In fact, the concept of a heartbeat is probably an unhelpful abstraction IMHO. You're effectively creating a table of durable locks, so this could be more general-purpose.

If we don't want to go too broad, maybe renaming this to `telemetry_locks` and explicitly name things like `locked_until`, `reported_until`, or something in that vein.